lib/ccf/index.ts

```diff
@@ -0,0 +1,259 @@
+import {IImpactModelInterface} from "../interfaces";
+import Spline from 'typescript-cubic-spline';
+import * as aws_instances from './aws-instances.json';
+import * as gcp_instances from './gcp-instances.json';
+import * as gcp_use from './gcp-use.json';
+import * as aws_use from './aws-use.json';
+import * as azure_use from './azure-use.json';
+import * as azure_instances from './azure-instances.json';
+import * as gcp_embodied from './gcp-embodied.json';
+import * as aws_embodied from './aws-embodied.json';
+import * as azure_embodied from './azure-embodied.json';
+
+interface IConsumption {
+    idle?: number;
+    tenPercent?: number;
+    fiftyPercent?: number;
+    hundredPercent?: number;
+    minWatts?: number;
+    maxWatts?: number;
+}
+
+interface IComputeInstance {
+    consumption: IConsumption;
+    embodiedEmission?: number;
+    name: string;
+    vCPUs?: number;
+    maxVCPUs?: number;
+}
+
+export class CloudCarbonFootprint implements IImpactModelInterface {
+    authParams: object | undefined;
+    // name of the data source
+    name: string | undefined;
+    // compute instances grouped by the provider with usage data
+    computeInstances: { [key: string]: { [key: string]: IComputeInstance } } = {};
+    gcpList: { [key: string]: any } = {};
+    azureList: { [key: string]: any } = {};
+    awsList: { [key: string]: any } = {};
+    provider: string = '';
+    instanceType: string = '';
+    expectedLifespan = 4;
+
+    constructor() {
+        this.standardizeInstanceMetrics();
+    }
+
+    authenticate(authParams: object): void {
+        this.authParams = authParams;
+    }
+
+    async configure(name: string, staticParams: object | undefined = undefined): Promise<IImpactModelInterface> {
```

Comment:
Every configuration param needs to be documented in the commend above the function (this needs to be the same format for every IMP). 
Eventually we need this data in some external file format, people shouldn't have to read code to understand how to configure a YAML file. But for now documenting the settings in the commend block above configure is fine.